pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/GrpcSendingMailbox.java

```diff
@@ -157,19 +167,28 @@ private StreamObserver<MailboxContent> getContentObserver() {
         .open(_statusObserver);
   }
 
-  private MailboxContent toMailboxContent(MseBlock block, List<DataBuffer> serializedStats)
+  private List<MailboxContent> toMailboxContents(MseBlock block, List<DataBuffer> serializedStats)
       throws IOException {
     _statMap.merge(MailboxSendOperator.StatKey.RAW_MESSAGES, 1);
     long start = System.currentTimeMillis();
     try {
       DataBlock dataBlock = MseBlockSerializer.toDataBlock(block, serializedStats);
-      ByteString byteString = DataBlockUtils.toByteString(dataBlock);
-      int sizeInBytes = byteString.size();
+      // so far we ensure payload is not bigger than maxBlockSize/2, we can fine tune this later
+      List<ByteString> byteStrings = DataBlockUtils.toByteStrings(dataBlock, getMaxBlockSize() / 2);
```

Comment:
`getMaxBlockSize()` won't change once the app is started, right? Do we need to read configs and parse strings to int once per block sent?